internal/aws/cloudwatch/go.mod

```diff
@@ -0,0 +1,33 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cloudwatch
```

Comment:
I think this module should be called `cloudwatchlogs`, or alternatively maybe have a `logs` package within `cloudwatch` but lean towards the former given how AWS documentation also has both as top level types

https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/Welcome.html
https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/Welcome.html

Anyways, just `cloudwatch` being logs code is misleading since Amazon CloudWatch is a metrics service, not a logs service

Reply:
@anuraaga so this package is going to be used to push cloudwatch metrics and also logs, so wouldn't it be appropriate to call it cloudwatch?

Reply:
The metrics are pushed via logs - this package I think is intended only for logs for now. Either way logs and metrics are too different to be in the same package, I feel the API docs make that clear.